File: Sharebook/Server/src/Services/Upload/Sharebook.Upload.API/Helpers/IHttpClientWebHelper.cs
namespace Sharebook.Upload.API.Helpers
{
    using System.Net.Http;
    using System.Threading.Tasks;

    using Sharebook.Upload.API.Web;
    using Sharebook.Upload.Application.Query;

    public interface IHttpClientWebHelper
    {
        Task<HttpResponseMessage> GetHttpResponseAsync(ITwitterQuery twitterQuery, ITwitterClientHandler handler = null);

        HttpClient GetHttpClient(ITwitterQuery twitterQuery, ITwitterClientHandler handler = null);
    }
}

File: Sharebook/Client/src/app/core/Core/Helpers/IHttpClientWebHelper.ts
import {inject, InjectionToken} from "@angular/core";
import {HttpClient, HttpResponse} from "@angular/common/http";

import {ITwitterQuery} from "../../Public/Models/Interfaces/ITwitterQuery";
import {ITwitterClientHandler} from "../Web/ITwitterClientHandler";
import {HttpClientWebHelper} from "../../../webLogic/HttpClientWebHelper";
import {OAuthWebRequestGeneratorFactory} from "../../../webLogic/OAuthWebRequestGenerator";
import {Observable} from "rxjs";

export interface IHttpClientWebHelper {
  getHttpResponseAsync(twitterQuery: ITwitterQuery, handler?: ITwitterClientHandler /*= null*/): Promise<Response>;

  getHttpClient(twitterQuery: ITwitterQuery, handler?: ITwitterClientHandler /*= null*/): HttpClient;
}

export const IHttpClientWebHelperToken = new InjectionToken<IHttpClientWebHelper>('IHttpClientWebHelper', {
  providedIn: 'root',
  factory: () => new HttpClientWebHelper(inject(OAuthWebRequestGeneratorFactory), inject(HttpClient)),
});
